anomaly_detector/core/job.py

```diff
@@ -0,0 +1,102 @@
+"""For all tasks that run in log anomaly detector you should implement the steps involved in train/infer of models."""
+import datetime
+import logging
+from abc import ABCMeta, abstractmethod
+import os
+from prometheus_client import Counter
+from anomaly_detector.model import W2VModel
+from anomaly_detector.adapters import FeedbackStrategy, SomModelAdapter, SomStorageAdapter
+from anomaly_detector.exception import EmptyDataSetException, ModelSaveException, ModelLoadException
+import time
+from opentracing_instrumentation.request_context import get_current_span, span_in_context
+
+TRAINING_COUNT = Counter("aiops_lad_train_count", "count of training runs")
+INFER_COUNT = Counter("aiops_lad_inference_count", "count of inference runs")
+
+
+class AbstractCommand(metaclass=ABCMeta):
+    """Generic command that gets executed when added to the TaskManager steps."""
+
+    @abstractmethod
+    def execute(self):
+        """Generic method that gets executed by the task manager."""
+        raise NotImplementedError("Please implement the <execute method>")
+
+
+class SomTrainJob(AbstractCommand):
+    """Som Training logic custom for each model. For this model we need to setup some configs."""
+
+    recreate_models = False
+
+    def __init__(self, node_map=24, model_adapter=None, recreate_model=True):
+        """Initialize inference job with fields to perform model training."""
```

Comment:
same `training job` here?

Reply:
Fixed the comments.

Reply:
Docstring says inference, you say training